internal/app/secretless/listeners/sshagent/listener.go

```
@@ -58,7 +59,20 @@ func (l *Listener) Listen() {
 		Resolver:      l.Resolver,
 	}
 
-	handler := l.RunHandlerFunc("sshagent", handlerOptions)
+	genericHandler := l.RunHandlerFunc("sshagent", handlerOptions)
+	// NOTE:
+	// This type coercion is a hack to prevent an even worse hack.  Originally, we had
+	// a method `LoadKeys` on the Handler interface itself, even though the method was
+	// only needed for this specific handler.  We decided that keeping the public
+	// interface clean was far more important, but in order to do that we must do this
+	// ugly type assertion in this handler.  Eventually, a more comprehensive refactor
+	// of the Secretless architecture will clean this up.
+	handler, ok := genericHandler.(*sshagent.Handler)
+	if !ok {
+		log.Printf("handler created is not for ssh-agent")
+		return
```

We definitely want to return an error I think if there's a problem